### PR TITLE
[SPARK-15115][SQL] Reorganize whole stage codegen benchmark suites

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?

We currently have a single suite that is very large, making it difficult to maintain and play with specific primitives. This patch reorganizes the file by creating multiple benchmark suites in a single package.

Most of the changes are straightforward move of code. On top of the code moving, I did:
1. Use SparkSession instead of SQLContext.
2. Turned most benchmark scenarios into a their own test cases, rather than having multiple scenarios in a single test case, which takes forever to run.
## How was this patch tested?

This is a test only change.

### CODE REVIEW

**Comment by rxin** [comment]
cc @davies

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/SortBenchmark.scala [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import org.apache.spark.util.random.XORShiftRandom
+
+/**
+ * Benchmark to measure performance for aggregate primitives.
+ * To run this:
+ *  build/sbt "sql/test-only *benchmark.SortBenchmark"
+ *
+ * Benchmarks in this file are skipped in normal builds.
+ */
+class SortBenchmark extends BenchmarkBase {
```
Comment (rxin) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/SortBenchmark.scala]: this was taken from RadixSortSuite

**Comment by davies** [comment]
LGTM

**Comment by rxin** [comment]
Thanks - merging in master/branch-2.0. The earlier test failure was unrelated.

**Pull request closed without merging**